test/commonControlTests/DynamicContainerControl.spec.ts

```diff
@@ -0,0 +1,195 @@
+        );
+        expect(response.prompt).equals('OK, Bob. And what is your last name, please?');
+    });
+});
+
+// -------------
+// Support classes for simple tests
+
+class DynamicNumbersControl extends DynamicContainerControl {
+    createDynamicChild(specification: DynamicControlSpecification): Control {
+        return new NumberControl({ id: specification.id });
+    }
+}
+
+// -------------
+// Support classes for e2e tests
+
+export class VariableControlsManager extends ControlManager {
+    public createControlTree(): Control {
+        const root = new ContainerControl({ id: 'root' });
+        root.addChild(new MyMultiControl({ id: 'multiValueContainer' }));
+        return root;
+    }
+}
+
+/**
+ * A custom container control that initially has no children. As the skill
+ * session progresses it adds one and then another child.
+ *
+ * The only information that needs to be tracked to recreate the controls is
+ * their id, hence the call to `addDynamicChild(specification = { id: 'firstName' })`.
+ *
+ * It is in createDynamicChild that the dynamic controls are actually created.
+ */
+export class MyMultiControl extends DynamicContainerControl {
+    /**
+     * a flag to help with prompt specialization. Note: this is not tracked in
+     * state as it does not need to persist between turns
+     */
+    wasTheFirstNameCapturedThisTurn: boolean;
+
+    async canHandle(input: ControlInput): Promise<boolean> {
+        let request = input.request;
+        if (request.type === 'LaunchRequest') {
+            return true;
+        }
+
+        request = request as IntentRequest; // assume IntentRequest.
+        const intent = request.intent;
+
+        const unpacked =
+            intent.name === 'GeneralControlIntent'
+                ? unpackGeneralControlIntent(intent)
+                : unpackSingleValueControlIntent(intent);
+
+        if (unpacked.action === 'addAnother') {
+            return true;
+        }
+
+        return this.canHandleByChild(input);
+    }
+
+    async handle(input: ControlInput, resultBuilder: ControlResultBuilder): Promise<void> {
+        let request = input.request;
+        if (request.type === 'LaunchRequest') {
+            // Dynamically add the first child
+            this.addDynamicChild({ id: 'firstName' });
+            return;
+        }
+        request = request as IntentRequest; // assume IntentRequest.
+        await this.handleByChild(input, resultBuilder);
+
+        if ((await this.children[0].isReady(input)) && this.children.length === 1) {
+            // Dynamically add the second child
+            this.addDynamicChild({ id: 'lastName' });
+
+            // And set a flag to tweak the next prompt a little
+            // Note, this type of this is not tracked in state
+            this.wasTheFirstNameCapturedThisTurn = true;
+        }
+        return;
+    }
+
+    createDynamicChild(spec: DynamicControlSpecification): Control {
```

Comment:
check that jsDoc defines that the constructed control matches the spec.id

Reply:
Are you adding any doc changes to handle this?

Reply:
oops forgot.. fixing